src/WarcraftLegacies.Source/Quests/Goblin/QuestRatchet.cs

```diff
@@ -0,0 +1,65 @@
+﻿using System.Collections.Generic;
+using MacroTools.ControlPointSystem;
+using MacroTools.Extensions;
+using MacroTools.FactionSystem;
+using MacroTools.LegendSystem;
+using MacroTools.ObjectiveSystem.Objectives.ControlPointBased;
+using MacroTools.ObjectiveSystem.Objectives.FactionBased;
+using MacroTools.ObjectiveSystem.Objectives.LegendBased;
+using MacroTools.ObjectiveSystem.Objectives.UnitBased;
+using MacroTools.QuestSystem;
+using static War3Api.Common;
+
+
+namespace WarcraftLegacies.Source.Quests.Goblin
+{
+  /// <summary>
+  /// The Goblins can acquire Ratchet.
+  /// </summary>
+  public sealed class QuestRatchet : QuestData
+  {
+    private readonly List<unit> _rescueUnits;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="QuestRatchet"/>.
+    /// </summary>
+    /// <param name="noggenfogger">Must be brought somewhere to complete the quest.</param>
+    public QuestRatchet(LegendaryHero noggenfogger) : base("Ratchet Port",
+      "The Port of Ratchet would be a great expansion of our trade empire.",
+      @"ReplaceableTextures\CommandButtons\BTNGoblinShop2.blp")
+    {
+      AddObjective(new ObjectiveControlPoint(ControlPointManager.Instance.GetFromUnitType(Constants.UNIT_N0A6_RATCHET)));
+      AddObjective(new ObjectiveLegendInRect(noggenfogger, Regions.Ratchet_Unlock_1, "Ratchet"));
+      AddObjective(new ObjectiveUpgrade(Constants.UNIT_O03N_FORTRESS_GOBLIN_T3, Constants.UNIT_O03L_GREAT_HALL_GOBLIN_T1));
+      AddObjective(new ObjectiveSelfExists());
+      Required = true;
+      ResearchId = Constants.UPGRADE_VQ01_QUEST_COMPLETED_RATCHET_PORT;
+      _rescueUnits = Regions.Ratchet_Unlock_1.PrepareUnitsForRescue(RescuePreparationMode.HideNonStructures,
+        filterUnit => filterUnit.GetTypeId() != FourCC("ngme"));
+    }
+
+    /// <inheritdoc />
+    protected override string RewardFlavour => "We have succesfully established a port at Ratchet!";
```

Comment:
It's fine, though it is a bit dull because it has no connection to the world. Is there any Ratchet-specific flavour we can use?

Reply:
Added additional flavour